ipam/manager_ipv6Ipam_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package ipam
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-container-networking/common"
+)
+
+var (
+
+	// Pools and addresses used by tests.
+	ipv6subnet1 = "ace:cab:deca:deed::/126"
+	ipv6addr1   = "ace:cab:deca:deed::1"
+	ipv6addr2   = "ace:cab:deca:deed::2"
+	ipv6addr3   = "ace:cab:deca:deed::3"
+)
+
+func createTestIpv6AddressManager() (AddressManager, error) {
+	var config common.PluginConfig
+	var options map[string]interface{}
+
+	options = make(map[string]interface{})
+	options[common.OptEnvironment] = common.OptEnvironmentIPv6Ipam
+
+	am, err := NewAddressManager()
+	if err != nil {
+		return nil, err
+	}
+
+	err = am.Initialize(&config, options)
+	if err != nil {
+		return nil, err
+	}
+
+	amImpl := am.(*addressManager)
+	src := amImpl.source.(*ipv6IpamSource)
+	src.nodeHostname = testNodeName
+	src.subnetMaskSizeLimit = testSubnetSize
+	src.kubeClient = newKubernetesTestClient()
```

Comment:
what is this? why do we have kube specific calls in unit tests

Reply:
Mocked out API server calls with mocked ClientSet k8s.io/client-go/kubernetes/fake